plugins/reach/batch/KSyncReachCreditTaskRunner/KSyncReachCreditTaskRunner.class.php

```diff
@@ -0,0 +1,80 @@
+<?php
+/**
+ * @package plugins.reach
+ * @subpackage Scheduler
+ */
+class KSyncReachCreditTaskRunner extends KPeriodicWorker
+{
+	/* (non-PHPdoc)
+	 * @see KBatchBase::getType()
+	 */
+	public static function getType()
+	{
+		return KalturaBatchJobType::SYNC_REACH_CREDIT_TASK;
+	}
+
+	/* (non-PHPdoc)
+	 * @see KBatchBase::getJobType()
+	 */
+	public function getJobType()
+	{
+		return self::getType();
+	}
+
+	/* (non-PHPdoc)
+	 * @see KBatchBase::run()
+	*/
+	public function run($jobs = null)
+	{
+		$reachClient = $this->SyncReachClient();
+		$filter = new KalturaReachProfileFilter();
+		$filter->statusEqual = KalturaReachProfileStatus::ACTIVE;
+		$pager = new KalturaFilterPager();
+		$pager->pageIndex = 1;
+		$pager->pageSize = 500;
+
+		$result = $reachClient->reachProfile->listAction($filter, $pager);
+		while ($result->totalCount > 0)
```

Comment:
Using that model we will have one extra call every time to the server.
The exsit from the loop should be when the returned amount is smaller than page size